yamux/src/connection/stream.rs

```diff
@@ -393,8 +415,9 @@ impl AsyncWrite for Stream {
         // a) to be consistent with outbound streams
         // b) to correctly test our behaviour around timing of when ACKs are sent. See `ack_timing.rs` test.
         if frame.header().flags().contains(ACK) {
-            self.shared()
-                .update_state(self.conn, self.id, State::Open { acknowledged: true });
+            self.shared.with_mut(|inner| {
+                inner.update_state(self.conn, self.id, State::Open { acknowledged: true });
+            });
```

Comment:
```suggestion
            self.shared.update_state(self.conn, self.id, State::Open { acknowledged: true });
```

Isn't there a `update_state` convenience function on `Shared`?

Reply:
Done.